Jellyfin.Server/Migrations/Routines/MigrateActivityLogDb.cs

```diff
@@ -0,0 +1,125 @@
+                { "Information", LogLevel.Information },
+                { "Info", LogLevel.Information },
+                { "Warn", LogLevel.Warning },
+                { "Warning", LogLevel.Warning },
+                { "Error", LogLevel.Error },
+                { "Critical", LogLevel.Critical }
+            };
+
+            var dataPath = _paths.DataPath;
+            using (var connection = SQLite3.Open(
+                Path.Combine(dataPath, DbFilename),
+                ConnectionFlags.ReadOnly,
+                null))
+            {
+                _logger.LogWarning("Migrating the activity database may take a while, do not stop Jellyfin.");
+                using var dbContext = _provider.CreateContext();
+
+                var queryResult = connection.Query("SELECT * FROM ActivityLog ORDER BY Id ASC");
+
+                // Make sure that the database is empty in case of failed migration due to power outages, etc.
+                dbContext.ActivityLogs.RemoveRange(dbContext.ActivityLogs);
+                dbContext.SaveChanges();
+                // Reset the autoincrement counter
+                dbContext.Database.ExecuteSqlRaw("UPDATE sqlite_sequence SET seq = 0 WHERE name = 'ActivityLog';");
+                dbContext.SaveChanges();
+
+                foreach (var entry in queryResult)
+                {
+                    if (!logLevelDictionary.TryGetValue(entry[8].ToString(), out var severity))
+                    {
+                        severity = LogLevel.Trace;
+                    }
+
+                    var newEntry = new ActivityLog(
+                        entry[1].ToString(),
+                        entry[4].ToString(),
+                        entry[6].SQLiteType == SQLiteType.Null ? Guid.Empty : Guid.Parse(entry[6].ToString()))
+                    {
+                        DateCreated = entry[7].ReadDateTime(),
+                        LogSeverity = severity
+                    };
+
+                    if (entry[2].SQLiteType != SQLiteType.Null)
+                    {
+                        newEntry.Overview = entry[2].ToString();
+                    }
+
+                    if (entry[3].SQLiteType != SQLiteType.Null)
+                    {
+                        newEntry.ShortOverview = entry[3].ToString();
+                    }
+
+                    if (entry[5].SQLiteType != SQLiteType.Null)
+                    {
+                        newEntry.ItemId = entry[5].ToString();
+                    }
+
+                    dbContext.ActivityLogs.Add(newEntry);
+                }
+
+                dbContext.SaveChanges();
+            }
+
+            try
+            {
+                File.Move(Path.Combine(dataPath, DbFilename), Path.Combine(dataPath, DbFilename + ".old"));
+            }
+            catch (IOException e)
+            {
+                _logger.LogError(e, "Error renaming legacy activity log database to 'activitylog.db.old'");
+            }
```

Comment:
We should probably move the journal file as well so it is picked up in case the user opens the renamed old database

```suggestion
            }
            
            try
            {
                var journalPath = Path.Combine(dataPath, DbFilename + "-journal");
                if (File.Exists(journalPath))
                {
                    File.Move(journalPath, Path.Combine(dataPath, DbFilename + ".old-journal"));
                }
            }
            catch (IOException e)
            {
                _logger.LogError(e, "Error renaming legacy activity log database journal to 'activitylog.db.old-journal'");
            }
```